wpilibj/src/main/java/edu/wpi/first/wpilibj/controller/ProfiledPIDController.java

```diff
@@ -319,10 +320,30 @@ public double calculate(double measurement, TrapezoidProfile.State goal,
   }
 
   /**
-   * Reset the previous error, the integral term, and disable the controller.
+   * Reset the previous error and the integral term.
+   * @param measurement the current measured State of the system.
    */
-  public void reset() {
+  public void reset(TrapezoidProfile.State measurement) {
     m_controller.reset();
+    m_setpoint = measurement;
+  }
+
+  /**
+   * Reset the previous error and the integral term.
+   * @param measuredPosition the current measured position of the system.
```

Comment:
How would I do that? Because the profiled pid controller doesn't have a defined unit -- it's whatever the user passes in -- and the Java State class is just `position` and `velocity`.

Reply:
The 2 DOF trajectory stuff does.